packages/http-client/src/http-client-types.ts

```diff
@@ -77,3 +88,29 @@ export interface HttpRequestOptions {
    */
   fetchCsrfToken?: boolean;
 }
+
+/**
+ * Origins of http request options. This indicates the priority of an http request option.
+ * Http request options with higher priorities will be used when reaching conflicts.
+ * The priority is "Custom" \> "DestinationProperty" \> "Destination" \> "RequestConfig"
+ */
+export type Origin =
+  | 'Custom'
+  | 'DestinationProperty'
+  | 'Destination'
+  | 'RequestConfig';
+
+/**
+ * @internal
+ */
+export type ValueWithOrigin = {
+  [key in Origin]?: string;
+};
+
+/**
+ * @internal
+ */
+export interface OptionWithOrigin {
```

Comment:
I found this additional type mislieading. I know it is convinient because you can get the options for a specific origin. However, you have to cast it back and fort and the `OptionWithOrigin` looks also like headers with origin from the type of `option`.